packages/ses/src/error/tame-v8-error-constructor.js

```diff
@@ -39,20 +39,45 @@ const safeV8CallSiteFacet = callSite => {
 
 const safeV8SST = sst => sst.map(safeV8CallSiteFacet);
 
-const callSiteFilter = _callSite => true;
-// const callSiteFilter = callSite =>
-//   !callSite.getFileName().includes('/node_modules/');
-
-const callSiteStringifier = callSite => `\n  at ${callSite}`;
-
-const stackStringFromSST = (error, sst) =>
-  [...sst.filter(callSiteFilter).map(callSiteStringifier)].join('');
+const FILENAME_FILTER = /^((?:.*[( ])?)[:/\w-_]*\/(packages\/.+)$/;
 
 export function tameV8ErrorConstructor(
   OriginalError,
   InitialError,
   errorTaming,
+  stackFiltering,
 ) {
+  // const callSiteFilter = _callSite => true;
+  const callSiteFilter = callSite => {
+    if (stackFiltering === 'verbose') {
+      return true;
+    }
+    const fileName = callSite.getFileName();
+    return (
+      !fileName ||
+      !(
+        fileName.includes('/node_modules/') ||
+        fileName.startsWith('internal/') ||
+        fileName.endsWith('/packages/ses/src/error/assert.js') ||
+        fileName.includes('/packages/eventual-send/src/')
+      )
+    );
+  };
+
+  const callSiteStringifier = callSite => {
+    let callSiteString = `${callSite}`;
+    if (stackFiltering === 'concise') {
+      const match = FILENAME_FILTER.exec(callSiteString);
+      if (match) {
+        callSiteString = `${match[1]}${match[2]}`;
```

Comment:
It seems to me that `match[2]` might be `undefined`.  We wouldn't want that to leak into the resulting string.  Please write a test case for that.

```suggestion
        callSiteString = `${match[1]}${match[2] || ''}`;
```

Reply:
Hi @michaelfig I'm happy to write a test case, but I don't understand. The pattern definition is
```js
const FILENAME_FILTER = /^((?:.*[( ])?)[:/\w-_]*\/(packages\/.+)$/;
```
I cannot see how a match could be successful without both `match[1]` and `match[2]` being strings. Can you think of a test case string for the pattern that could produce the outcome you're worried about?

Reply:
I added a comment above `FILENAME_FILTER` to clarify.

I also found out how to do collapsible sections in markdown, and used it to add some example log output to the `lockdown-options.md` page. Collapsible sections are cool!

Reply:
Okay, that was my misreading.  Testing it shows it doesn't have the problem I was worried about.